inspire/modules/migrator/tasks.py

```diff
@@ -22,10 +22,100 @@
 
 """Manage migration from INSPIRE legacy instance."""
 
+from __future__ import absolute_import, print_function
+
+from celery.utils.log import get_task_logger
+
+import gzip
+
+from dojson.contrib.marc21.utils import create_record as marc_create_record
+from dojson.contrib.marc21.utils import split_stream
 from invenio_celery import celery
+from invenio_records.api import Record as record_api
+from invenio_records.models import Record
+from lxml import etree
+
+from inspire.dojson.conferences import conferences
+from inspire.dojson.experiments import experiments
+from inspire.dojson.hep import hep
+from inspire.dojson.hepnames import hepnames
+from inspire.dojson.institutions import institutions
+from inspire.dojson.jobs import jobs
+from inspire.dojson.journals import journals
+from inspire.dojson.processors import _collection_in_record
+from invenio_ext.sqlalchemy import db
+
+logger = get_task_logger(__name__)
+
+CHUNK_SIZE = 1000
+
+
+def chunker(iterable, chunksize):
+    buf = []
+    for elem in iterable:
+        buf.append(elem)
+        if len(buf) == chunksize:
+            yield buf
+            buf = []
+    if buf:
+        yield buf
 
 
 @celery.task
-def migrate(url, **kwargs):
+def migrate(source, **kwargs):
     """Main migration function."""
-    pass
+    broken_output = kwargs.get('broken_output')
```

Comment:
Instead of getting these from the kwargs, it would be better to just add them to the function signature. Say:

```def migrate(source, broken_output=None, dry_run=False, **kwargs):```

PS: Looks also like `kwargs` will be unused then or?